src/anm/media/video.js

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2011-@COPYRIGHT_YEAR by Animatron.
+ * All rights are reserved.
+ *
+ * Animatron Player is licensed under the MIT License, see LICENSE.
+ *
+ * @VERSION
+ */
+
+var conf = require('../conf.js'),
+    log = require('../log.js');
+
+var C = require('../constants.js');
+
+var engine = require('engine');
+
+var ResMan = require('../resource_manager.js');
+
+function Video(url) {
+    this.url = url;
+    this.ready = false;
+    this.playing = false;
+}
+
+Video.prototype.connect = function(element) {
+    var me = this;
+    element.on(C.X_START, function() {
+        me.play.apply(me, arguments);
+    });
+    var stop = function() { me.stop(); };
+    element.on(C.X_STOP, stop);
+    element.on(C.S_STOP, stop);
+    element.on(C.S_PAUSE, stop);
+};
+Video.prototype.load = function(player) {
+
+    var me = this;
+    ResMan.loadOrGet(player.id, me.url,
+        function(notify_success, notify_error, notify_progress) { // loader
+            var url = me.url;
+            if (engine.isHttps) { url = url.replace('http:', 'https:'); }
+
+            var el = engine.createVideo();
+            el.setAttribute("preload", "auto");
+            el.style.display = 'none';
+
+            var progressListener = function(e) {
+                var buffered = el.buffered;
+                if (buffered.length == 1) {
+                    if (el.readyState === 4) {
```

Comment:
[`HAVE_ENOUGH_DATA`](https://developer.mozilla.org/en/docs/Web/API/HTMLMediaElement). Yeah, may be it should be moved to constant or a have a comment there or even be a part of enum or..